### PR TITLE
feat(sidebar): add styled system to sidebar - FE-3574

### PR DESCRIPTION
**Pull request by bulaj**
BREAKING CHANGE: SidebarHeader is now internal component, use header prop

### Proposed behaviour
Moved SidebarHeader to internals.
Sidebar's header can be supplied using ``header`` prop (accepts node).
Sidebar's content will have visible scroll when exceeding viewport.
image

### Current behaviour
SidebarHeader is available as a component and can be used as a child of Sidebar.
Sidebar content has no scroll if content exceeds the viewport.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] <del>Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] <del>Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context
There are two known issues with stories:
- prop table does not show correct props,
- issues with state implemenation which causes only default example to be automatically open when navigating.
Those are already on backlog.

### Testing instructions
By examples available in storybook, search for Sidebar.

### CODE REVIEW

File: src/components/sidebar/__internal__/sidebar-header/sidebar-header.style.js [hunk 1]
```diff
@@ -1,21 +1,24 @@
 import styled from "styled-components";
-import baseTheme from "../../../style/themes/base";
+import { padding } from "styled-system";
+import baseTheme from "../../../../style/themes/base";
 
 const SidebarHeaderStyle = styled.div`
+  ${padding};
```
Comment (grabkowski) [inline src/components/sidebar/__internal__/sidebar-header/sidebar-header.style.js:6]: I can't find where these padding props are passed to this component. Looks like they are only defined in `defaultProps`.

File: src/components/sidebar/sidebar.style.js [hunk 2]
```diff
@@ -13,12 +15,12 @@ const sidebarSizes = {
 };
 
 const SidebarStyle = styled.div`
+  ${padding};
```
Comment (grabkowski) [inline src/components/sidebar/sidebar.style.js:18]: the same as with `SidebarHeaderStyle`

File: src/components/sidebar/__internal__/sidebar-header/sidebar-header.style.js (hunk 1 above)
Comment (sjohnson-sage) [reply src/components/sidebar/__internal__/sidebar-header/sidebar-header.style.js:6]: Yeah you need to add the padding props into `SidebarHeader` - similar to how the margin props are added to https://github.com/Sage/carbon/blob/master/src/__experimental__/components/radio-button/radio-button.component.js

File: src/components/sidebar/sidebar.component.js [hunk 3]
```diff
@@ -82,6 +85,8 @@ Sidebar.propTypes = {
   position: PropTypes.string,
   /** Sets the size of the sidebar when open. */
   size: PropTypes.string,
+  /** Node that will be used as sidebar header. */
+  header: PropTypes.node,
```
Comment (sjohnson-sage) [inline src/components/sidebar/sidebar.component.js:89]: We need to add the styled-system padding props to the propTypes here as well and make sure they are passed down to `SidebarStyle`

File: src/components/sidebar/sidebar.style.js [hunk 4]
```diff
@@ -42,8 +44,20 @@ const SidebarStyle = styled.div`
   }
 `;
 
+export const SidebarContentBox = styled(Box)``;
+
+SidebarContentBox.defaultProps = {
+  mr: -4,
```
Comment (sjohnson-sage) [inline src/components/sidebar/sidebar.style.js:50]: There must be a better way to do this than using negative margins - we really should avoid using negative margins

File: src/components/sidebar/sidebar.style.js [hunk 5]
```diff
@@ -42,8 +44,20 @@ const SidebarStyle = styled.div`
   }
 `;
 
+export const SidebarContentBox = styled(Box)``;
```
Comment (sjohnson-sage) [inline src/components/sidebar/sidebar.style.js:47]: This whole styled component is unnecessary, we could just have 
```jsx
<Box 
  mr={-4}
  pr={4}
  height="calc(100% - 64px)"
  scrollVariant= "light"
  overflow="auto"
>
  {children} 
</Box>
```

in `sidebar.component.js`. The user can't override any of these so they don't need to be default props, just hardcoded ones

**Review by sjohnson-sage — COMMENTED** [review]
Just a few comments from me on top of Kuba's - could you refactor the CSS to avoid using negative margins please, now seems like a good opportunity to do that

**Review by sjohnson-sage — APPROVED** [review]

**Review by grabkowski — DISMISSED** [review]

**Review by sjohnson-sage — DISMISSED** [review]

File: src/components/sidebar/sidebar.component.js [hunk 6]
```diff
@@ -48,17 +50,28 @@ class Sidebar extends Modal {
...
-        position={this.props.position}
-        size={this.props.size}
+        position={position}
+        size={size}
         data-element="sidebar"
       >
         {this.closeIcon()}
-        {this.props.children}
+        {header && <SidebarHeader>{header}</SidebarHeader>}
+        {/* Box for sideboard content */}
```
Comment (DlgSHi) [inline src/components/sidebar/sidebar.component.js:65]:
```suggestion
        {/* Box for sidebad content */}
```

File: src/components/sidebar/sidebar.stories.mdx [hunk 7]
```diff
@@ -30,58 +30,92 @@ To position the sidebar on the left hand side pass `position='left'` to the comp
...
+            <div style={{ marginBottom: 3000 }}>Long content</div>
+          </Sidebar>
+        </>
+      );
+    }}
+  </Story>
+</Preview>
 
-### SidebarHeader
+## Props:
```
Comment (DlgSHi) [inline src/components/sidebar/sidebar.stories.mdx:119]:
```suggestion
## Props
```

**Review by DlgSHi — COMMENTED** [review]
few small changes needed

File: src/components/sidebar/sidebar.stories.mdx [hunk 8]
```diff
@@ -30,58 +30,92 @@ To position the sidebar on the left hand side pass `position='left'` to the comp
...
+        </>
+      );
+    }}
+  </Story>
+</Preview>
+
+### With header
+
+<Preview>
+  <Story name="with-header" parameters={{ chromatic: { disable: true } }}>
```
Comment (DlgSHi) [inline src/components/sidebar/sidebar.stories.mdx:64]:
```suggestion
  <Story name="with header" parameters={{ chromatic: { disable: true } }}>
```

File: src/components/sidebar/sidebar.stories.mdx [hunk 9]
```diff
@@ -30,58 +30,92 @@ To position the sidebar on the left hand side pass `position='left'` to the comp
...
 
-## Props:
+### With scroll
 
-### Sidebar
+If sidebar content does not fit the sidebar, scroll will appear. Default scroll theme is "light".
 
-<Props of={Sidebar} />
+<Preview>
+  <Story name="with-scroll" parameters={{ chromatic: { disable: true } }}>
```
Comment (DlgSHi) [inline src/components/sidebar/sidebar.stories.mdx:94]:
```suggestion
  <Story name="with scroll" parameters={{ chromatic: { disable: true } }}>
```

**Review by DlgSHi — COMMENTED** [review]
Remove also the `header` from `Test -> default` story

**Review by grabkowski — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Pull request merged**